pkgs/development/python-modules/mkdocs/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, python3
+, fetchFromGitHub
+, buildPythonPackage
+, fetchpatch
+, click
+, jinja2
+, markdown
+, mergedeep
+, pyyaml
+, pyyaml-env-tag
+, ghp-import
+, importlib-metadata
+, watchdog
+, packaging
+, Babel
+, mock
+, pytestCheckHook
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs";
+  version = "1.2.3";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-LBw2ftGyeNvARQ8xiYUho8BiQh+aIEqROP51gKvNxEo=";
+  };
+
+  patches = [
+    # TODO: remove this patch on the next release
+    # the upstream patch https://github.com/mkdocs/mkdocs/commit/c93fc91e4dc0ef33e2ea418aaa32b0584a8d354a.patch
+    # should work, but doesn't because tox.ini doesn't apply cleanly
+    # and fetchpatch with excludes doesn't seem to have any effect
+    ./fix-tests-no-tox-ini.patch
+  ];
+
+  propagatedBuildInputs = [
+    click
+    jinja2
+    markdown
+    mergedeep
+    pyyaml
+    pyyaml-env-tag
+    ghp-import
+    importlib-metadata
+    watchdog
+    packaging
+  ];
+
+  checkInputs = [
+    Babel
+    mock
+    pytestCheckHook
+  ];
+
+  pytestFlagsArray = [ "mkdocs/tests/*.py" ];
```

Comment:
```suggestion
```

pytest is smart enough to pick this up.

Reply:
It's not though, because the tests do not start with `test_*.py`. All the test in mkdocs are named `*_tests.py`